usig/sgx/usig-enclave.go

```diff
@@ -16,12 +16,69 @@
 
 package sgx
 
-// #cgo pkg-config: $SGX_URTS_PACKAGE
-// #cgo CFLAGS: -I${SRCDIR}/shim
-// #cgo LDFLAGS: -L${SRCDIR}/shim -lusig_shim
-//
-// #include <stdlib.h>
-// #include "usig.h"
+/*
+// TODO Use pkg-config?
+#cgo CPPFLAGS: -I/opt/intel/sgxsdk/include
+#cgo LDFLAGS: -ldl
+
+#include <dlfcn.h>
+#include <stdlib.h>
+
+#include <sgx_urts.h>
+#include <sgx_tcrypto.h>
+
+// wrappers to call a function through a function pointer, which is returned by
+// dlsym().
+
+sgx_status_t usig_init(const void *pointer,
+                       const char *enclave_file, sgx_enclave_id_t *enclave_id,
+                       void *sealed_data, size_t sealed_data_size)
+{
+	return ((sgx_status_t (*)(const char *, sgx_enclave_id_t *, void *, size_t)) pointer)
+			(enclave_file, enclave_id, sealed_data, sealed_data_size);
```

Comment:
It might be a bit more convenient to define a function type:

```c
typedef usig_init_fn(const void *pointer,
                     const char *enclave_file, sgx_enclave_id_t *enclave_id,
                     void *sealed_data, size_t sealed_data_size)
```

So the result of `dlsym` can be assigned to a variable of that type. Then the variable could be invoked as normal function.